The Mac Weekly/PostViewController.swift

```diff
@@ -108,9 +108,25 @@ class PostViewController: UIViewController, UIScrollViewDelegate, WKNavigationDe
             fatalError("Requires Post")
         }
         postTitleLabel.text = post.title
-        let formatter = DateFormatter()
-        formatter.dateFormat = defaultDateFormat
-        postDateLabel.text = formatter.string(from: post.time)
+        
+        // Format the date nicely - Gabriel Brown
+        let postTimeInterval = -(post.time.timeIntervalSinceNow)  // Minus sign in front to flip the negative, since everything happened in the past
+        let timeUnit = TestableUtils.getTimeUnitToUse(timeInterval: postTimeInterval)
+        
+        if timeUnit != nil {
+            let formatter = DateComponentsFormatter()
+            formatter.unitsStyle = .short
+            formatter.includesTimeRemainingPhrase = false
+            formatter.allowedUnits = timeUnit!      // I feel like there should be a better way to do this but idk how. At least I'm explicitly checking that it's not nil?
+            
+            postDateLabel.text = formatter.string(from: postTimeInterval)! + " ago"
+        }
+        else {
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateStyle = .long
+            postDateLabel.text = dateFormatter.string(from: post.time)
+        }
+        
```

Comment:
Could we refactor this block of code out to a function in Utils, since it gets repeated the same way twice in these two files?

Reply:
It depends on whether or not we want the date to be displayed the same way in both the table view and post view. Right now, I have the date in the table view set as short and the date in the post view set to long. I feel like if we want them to be different then it might be more trouble than it's worth to try and work it all into Utils, but let me know what you think.

Reply:
Hmm... maybe its ok to leave for now

I was going to suggest a boolean flag for long or short dates, but im not sure what we want out of that in the future. This code seems simple enough and it seems like it'd actually be appropriate for it to be custom. 

It might make sense to refactor the `timeUnit != nil` case into a `displayDateInterval` sort of function (not sure about that function name)